src/AI/WallyAI.java

```diff
@@ -190,6 +190,30 @@ else if( unitQueue.isEmpty() )
       }
 
       Queue<Unit> tempQueue = new ArrayDeque<Unit>();
+
+      // Queue all of the capture actions we don't have to move for
+      while (actions.isEmpty() && !unitQueue.isEmpty())
+      {
+        Unit unit = unitQueue.poll();
+        XYCoord position = new XYCoord(unit.x, unit.y);
+
+        if( unit.getCaptureProgress() > 0 )
+        {
+          actions.offer(new GameAction.CaptureAction(gameMap, unit, Utils.findShortestPath(unit, position, gameMap)));
+        }
+        else
+        {
+          tempQueue.offer(unit);
```

Comment:
Why not just do a peek, and then do a poll if you create the action instead of shoving them into a temp queue and back again?

Reply:
Because I wasn't paying very good attention to the loop condition I copied.
My intent was to queue all such actions at once.